util/security/secret.py

```diff
@@ -22,8 +22,9 @@ def convert_secret_key(config_secret_key):
             pass
 
     if secret_key is None:
-        secret_key = str(bytearray(list(map(ord, config_secret_key))))
+        secret_key = bytearray(list(map(ord, config_secret_key)))
```

Comment:
This is causing differences in `test/data/test.db` when running in Python 2 versus Python 3. Investigating.